api/server/services/ToolService.js

```
@@ -430,60 +488,79 @@ async function loadAgentTools({ req, agent_id, tools, tool_resources, openAIApiK
-        if (toolName.includes(domain)) {
-          currentDomain = domain;
-          actionSet = action;
-          break;
-        }
+    if (!actionSets.length) {
+      actionSets = (await loadActionSets({ agent_id: agent.id })) ?? [];
+    }
+
+    let actionSet = null;
+    let currentDomain = '';
+    for (let action of actionSets) {
```

Biome (1.9.4)

[error] This let declares a variable that is only assigned once.

'action' is never reassigned.

Safe fix: Use const instead.


(lint/style/useConst)